ui/src/tempVars/components/MetaQueryTemplateBuilder.tsx

```diff
@@ -0,0 +1,168 @@
+import React, {PureComponent, ChangeEvent} from 'react'
+import _ from 'lodash'
+
+import {proxy} from 'src/utils/queryUrlGenerator'
+import {ErrorHandling} from 'src/shared/decorators/errors'
+import TemplateMetaQueryPreview from 'src/tempVars/components/TemplateMetaQueryPreview'
+import {parseMetaQuery, isInvalidMetaQuery} from 'src/tempVars/utils/parsing'
+
+import {
+  TemplateBuilderProps,
+  RemoteDataState,
+  TemplateValueType,
+} from 'src/types'
+
+const DEBOUNCE_DELAY = 750
+
+interface State {
+  metaQueryInput: string // bound to input
+  metaQuery: string // debounced view of metaQueryInput
+  metaQueryResults: string[]
+  metaQueryResultsStatus: RemoteDataState
+}
+
+@ErrorHandling
+class CustomMetaQueryTemplateBuilder extends PureComponent<
+  TemplateBuilderProps,
+  State
+> {
+  private handleMetaQueryChange: () => void = _.debounce(() => {
+    const {metaQuery, metaQueryInput} = this.state
+
+    if (metaQuery === metaQueryInput) {
+      return
+    }
+
+    this.setState({metaQuery: metaQueryInput}, this.executeQuery)
+  }, DEBOUNCE_DELAY)
+
+  constructor(props) {
+    super(props)
+
+    const metaQuery = _.get(props.template, 'query.influxql', '')
```

Comment:
could `getDeep` be used here instead of `_.get`? it uses generics and you can pass in the type for it to fall back to.